src/core/algorithms/md/hymd/lattice/md_lattice.cpp

```diff
@@ -586,6 +617,9 @@ void MdLattice::MdVerificationMessenger::LowerAndSpecialize(
         rhs.Set(rhs_index, new_ccv_id);
     }
     lattice_->Specialize(GetLhs(), invalidated.GetInvalidated());
+    if (GetRhs().IsEmpty() && GetNode().IsEmpty()) {
+        lattice_->TryDeleteEmptyNode(GetLhs());
```

Comment:
We know for certain that the node exists at this point, check on line 580 is extraneous.